mirascope/base/embedders.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar, Generic, Optional, TypeVar
+
+from pydantic import BaseModel
+
+from .types import BaseEmbeddingParams
+
+BaseEmbeddingT = TypeVar("BaseEmbeddingT", bound=BaseModel)
+
+
+class BaseEmbedder(BaseModel, Generic[BaseEmbeddingT], ABC):
+    api_key: ClassVar[Optional[str]] = None
+    base_url: ClassVar[Optional[str]] = None
+    embedding_params: ClassVar[BaseEmbeddingParams] = BaseEmbeddingParams(
```

Comment:
this should either be `embedder_params` and `BaseEmbedderParams` to match `chunker_params` and `BaseChunkerParams` or we should switch to use `ChunkingParams` and `chunking_params`

I prefer the latter.

Reply:
okay I can switch to ChunkingParams